Documentation/Upgrade/rook-upgrade.md

```diff
@@ -14,14 +14,15 @@ We welcome feedback and opening issues!
 
 ## Supported Versions
 
-This guide is for upgrading from **Rook v1.8.x to Rook v1.9.x**.
```

Comment:
In #10744 I had made a first pass at the upgrade guide, but that was only directly in the 1.10 branch unfortunately. To avoid merge conflicts, we'll likely just need to use this PR for master, and open a separate PR for the 1.10 branch.

Reply:
Instead, I cherry-picked the commit from that PR to bring master back in line so we can keep doing doc backports more easily.